src/python/pants/init/extension_loader.py

```diff
@@ -155,21 +217,33 @@ def invoke_entrypoint(name: str):
                 f"Entrypoint {name} in {backend_module} must be a zero-arg callable: {e!r}"
             )
 
-    target_types = invoke_entrypoint("target_types")
-    if target_types:
-        build_configuration.register_target_types(backend_package, target_types)
-    build_file_aliases = invoke_entrypoint("build_file_aliases")
-    if build_file_aliases:
-        build_configuration.register_aliases(build_file_aliases)
-    rules = invoke_entrypoint("rules")
-    if rules:
-        build_configuration.register_rules(backend_package, rules)
-    remote_auth_func = getattr(module, "remote_auth", None)
-    if remote_auth_func:
-        logger.debug(
-            f"register remote auth function {remote_auth_func.__module__}.{remote_auth_func.__name__} from backend: {backend_package}"
+    if hasattr(module, _EXTENSION_INITIALIZE_METHOD_NAME):
+        initializer = getattr(module, _EXTENSION_INITIALIZE_METHOD_NAME)
+        initializer_context = PluginExtensionInitContext(
+            key=backend_package, build_configuration=build_configuration
         )
-        build_configuration.register_remote_auth_plugin(remote_auth_func)
-    auxiliary_goals = invoke_entrypoint("auxiliary_goals")
-    if auxiliary_goals:
-        build_configuration.register_auxiliary_goals(backend_package, auxiliary_goals)
+        try:
+            initializer(initializer_context)
+        except Exception as e:
+            raise BackendConfigurationError(
+                f"Backend {backend_package} failed to initialize: {e!r}"
+            ) from e
+    else:
+        target_types = invoke_entrypoint("target_types")
```

Comment:
Not for this PR: Find a way to fold this logic into just being able to call `_legacy_extension_init`.